src/mono/mono/metadata/class-init.c

```diff
@@ -2416,6 +2459,9 @@ mono_class_layout_fields (MonoClass *klass, int base_instance_size, int packing_
 		class_size = field_offsets [i] + size;
 	}
 
+	if (m_class_is_inlinearray (klass) && has_static_fields)
+		mono_class_set_type_load_failure (klass, "Struct with inline array atribute must have a single field.");
```

Comment:
Will we get a TLE for something like

```csharp
[InlineArray(10)]
struct EmptyStruct
{
}
```

Also, is there a reason why we're not allowing `static` fields in an InlineArray struct?  That seems unrelated.  Static fields don't really affect the layout at all, and I could imagine they could be useful (for example to cache some delegates or something).

Reply:
According to the test, static fields shouldn't be inlined, but you are right that it is not handled properly. Added a check  for empty structs in `mono_class_layout_fields`.

https://github.com/dotnet/runtime/blob/eefaa5a36af292a99be4e4421cb17e56e630f6cd/src/tests/Loader/classloader/InlineArray/InlineArrayInvalid.cs#L100-L116